sdk/cosmos/azure-cosmos/azure/cosmos/_change_feed/change_feed_fetcher.py

```diff
@@ -0,0 +1,199 @@
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""Internal class for processing change feed implementation in the Azure Cosmos
+database service.
+"""
+import base64
+import copy
+import json
+from abc import ABC, abstractmethod
+from typing import Dict, Any, List
+
+from azure.cosmos import _retry_utility, http_constants, exceptions
+from azure.cosmos._change_feed.change_feed_start_from import ChangeFeedStartFromPointInTime
+from azure.cosmos._change_feed.change_feed_state import ChangeFeedStateV1, ChangeFeedStateV2
+from azure.cosmos.exceptions import CosmosHttpResponseError
+
+# pylint: disable=protected-access
+
+class ChangeFeedFetcher(ABC):
+
+    @abstractmethod
+    def fetch_next_block(self):
+        pass
+
+class ChangeFeedFetcherV1(ChangeFeedFetcher):
+    """Internal class for change feed fetch v1 implementation.
+     This is used when partition key range id is used or when the supplied continuation token is in just simple etag.
+     Please note v1 does not support split or merge.
+
+    """
+    def __init__(
+            self,
+            client,
+            resource_link: str,
+            feed_options: Dict[str, Any],
+            fetch_function):
+
+        self._client = client
+        self._feed_options = feed_options
+
+        self._change_feed_state: ChangeFeedStateV1 = self._feed_options.pop("changeFeedState")
+        if not isinstance(self._change_feed_state, ChangeFeedStateV1):
+            raise ValueError(f"ChangeFeedFetcherV1 can not handle change feed state version"
+                             f" {type(self._change_feed_state)}")
+
+        self._resource_link = resource_link
+        self._fetch_function = fetch_function
+
+    def fetch_next_block(self):
+        """Returns a block of results.
+
+        :return: List of results.
+        :rtype: list
+        """
+        def callback():
+            return self.fetch_change_feed_items(self._fetch_function)
+
+        return _retry_utility.Execute(self._client, self._client._global_endpoint_manager, callback)
+
+    def fetch_change_feed_items(self, fetch_function) -> List[Dict[str, Any]]:
```

Comment:
I wonder what prevents us from reusing the code in `aio` and outside of `aio` module.

Reply:
the logic flow is pretty similar, but mainly different on the io calls with async semantic vs sync semantic